src/accelerate/data_loader.py

```diff
@@ -650,12 +650,12 @@ def prepare_data_loader(
     # Iterable dataset doesn't like batch_sampler, but data_loader creates a default one for it
     new_batch_sampler = dataloader.batch_sampler if not isinstance(new_dataset, IterableDataset) else None
     sampler_is_batch_sampler = False
-    generator = getattr(dataloader, "generator", None)
+    synchronized_generator = getattr(dataloader, "generator", None)
```

Comment:
We should probably leave this one out and pass it as the generator (as initially suggested in your PR).

Reply:
"generator" is removed from `ignore_kwargs`.

Reply:
Yes, but this one should not be synchronized I think. Or do we want to default to the dataloader generator if tehre is no generator in the sampler?

Reply:
Sorry I didn't get your point. Yes, this line should be `synchronized_generator = None`, as #790 .

Reply:
Oops! Yup, resolved the comment